src/formatDate.js

```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatedDate = [];
+  const partsDate = {};
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const initialDate = date.split(oldSeparator);
+
+  fromFormat.forEach((partDate, index) => {
+    partsDate[partDate] = initialDate[index];
+  });
+
+  if (partsDate.hasOwnProperty('YY')) {
+    partsDate['YYYY'] = (partsDate['YY'] < 30)
+      ? '20' + partsDate['YY']
+      : '19' + partsDate['YY'];
```

Comment:
Use template strings instead of string concatenation for forming the 'YYYY' part.